.github/workflows/validate-github-access.yml

```diff
@@ -0,0 +1,313 @@
+          echo "=================================="
+          echo "GitHub Base URL: ${{ inputs.github_base_url }}"
+          echo "Organization: ${{ inputs.github_org }}"
+          echo "Test Repository: ${{ inputs.test_repo }}"
+          echo "Test Version: ${{ inputs.test_version }}"
+          echo "Token Secret: ${{ inputs.github_token_secret }}"
+          echo ""
+          
+          # Validate URL format
+          if [[ "${{ inputs.github_base_url }}" != http* ]]; then
+            echo "❌ ERROR: GitHub Base URL must start with http:// or https://"
+            exit 1
+          fi
+
+      - name: Test Repository Access
+        env:
+          GH_TOKEN: ${{ secrets[inputs.github_token_secret] }}
+          GITHUB_BASE_URL: ${{ inputs.github_base_url }}
+          GITHUB_ORG: ${{ inputs.github_org }}
+          REPO_NAME: ${{ inputs.test_repo }}
+          TEST_VERSION: ${{ inputs.test_version }}
+        run: |
+          echo "🔍 Testing Repository Access"
+          echo "============================"
+          
+          # Check if token is provided
+          if [ -z "$GH_TOKEN" ]; then
+            echo "❌ ERROR: GitHub token not found in secrets.${{ inputs.github_token_secret }}"
+            echo ""
+            echo "💡 SOLUTION: Add your GitHub token to repository secrets with name '${{ inputs.github_token_secret }}'"
+            echo "   For GitHub.com: Create a Personal Access Token with 'repo' scope"
+            echo "   For GitHub Enterprise: Create a token with 'repo' scope on your enterprise instance"
+            exit 1
+          fi
+          
+          # Determine API URL (same logic as patcher-action)
+          if [ "$GITHUB_BASE_URL" = "https://github.com" ]; then
+            API_URL="https://api.github.com"
+            echo "🌐 Using GitHub.com API: $API_URL"
+          else
+            API_URL="${GITHUB_BASE_URL}/api/v3"
+            echo "🏢 Using GitHub Enterprise API: $API_URL"
+          fi
+          
+          echo ""
+          echo "📋 Testing access to: ${GITHUB_BASE_URL}/${GITHUB_ORG}/${REPO_NAME}"
+          
+          # Test repository access
+          response=$(curl -s -w "%{http_code}" \
+            -H "Authorization: Bearer ${GH_TOKEN}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            -H "User-Agent: patcher-action-validator" \
+            "${API_URL}/repos/${GITHUB_ORG}/${REPO_NAME}" \
+            -o /tmp/repo_response.json)
+          
+          http_code="${response: -3}"
+          echo "📡 Repository API Response: $http_code"
+          
+          if [ "$http_code" = "200" ]; then
+            echo "✅ SUCCESS: Repository access granted!"
+            echo ""
+            echo "📊 Repository Details:"
+            if command -v jq >/dev/null 2>&1; then
+              cat /tmp/repo_response.json | jq -r '"  Name: " + .name, "  Full Name: " + .full_name, "  Private: " + (.private | tostring), "  Default Branch: " + .default_branch'
```

Comment:
NIT: Unnecessary `cat`. You can use redirection instead.

```
jq ... < /tmp/repo_response.json
```

Reply:
Ignoring due to low importance.